app/services/recovery_code_generator.rb

```diff
@@ -10,12 +10,33 @@ def create
     raw_recovery_code
   end
 
+  def verify(plaintext_code)
+    uak = user_access_key(plaintext_code)
+    encryption_key, encrypted_code = user.recovery_code.split(Pii::Encryptor::DELIMITER)
+    begin
+      key_maker.unlock(uak, encryption_key)
+    rescue Pii::EncryptionError => _err
+      return false
```

Comment:
Maybe? We don't log failed password verification attempts afaik. This would be analogous to `User.valid_password?` or `OpenSSL::PKey::RSA.verify`. Those don't throw an exception, they just return booleans. The reason we catch the exception here is that verification requires decrypting the `uak` and we will get invalid base64 or OpenSSL errors that bubble up.